dbms/src/Server/Server.cpp

```diff
@@ -1136,7 +1138,14 @@ int Server::main(const std::vector<std::string> & /*args*/)
     global_context->initializeGlobalStoragePoolIfNeed(global_context->getPathPool());
     LOG_INFO(log, "Global PageStorage run mode is {}", static_cast<UInt8>(global_context->getPageStorageRunMode()));
 
-    global_context->initializeWriteNodePageStorageIfNeed(global_context->getPathPool());
+    if (global_context->isDisaggregatedStorageMode())
+        global_context->initializeWriteNodePageStorageIfNeed(global_context->getPathPool());
```

Comment:
1. Now if user want to use S3, they won't set `flash.disaggregated_mode`
2. `isDisaggregatedStorageMode` can not distinguish "non-disagg tiflash instance" or "disagg-write-node tiflash instance" either
https://github.com/pingcap/tiflash/blob/8794db4ebc003d8a12e02bf0d2aa889f84ab7a19/dbms/src/Interpreters/Context.h#L515-L519

Reply:
`isDisaggregatedStorageMode` can not distinguish "non-disagg tiflash instance" or "disagg-write-node tiflash instance" is by design